packages/suggestion/src/suggestion.ts

```diff
@@ -247,9 +253,14 @@ export function Suggestion<I = any, TSelected = any>({
           }
 
           if (handleChange || handleStart) {
-            props.items = await items({
-              editor,
-              query: state.query,
+            props.isLoading = true
+            Promise.resolve(items({ editor, query: state.query })).then(result => {
+              if (props == null) {
+                return
+              }
+              props.isLoading = false
+              props.items = result
+              renderer?.onUpdate?.(props)
```

Comment:
This is doing more than just exposing the loading state, it should still be awaited, you are changing the behavior of how this works.

Reply:
Ah I see apologies - this is kind of a naive question, why does it need to be awaited?

Reply:
By awaiting the function, you are suspending it's execution of the lines after it. If the goal truly is to only expose an `isLoading` flag, then you should not be messing with the order of execution. By doing a Promise.resolve & not awaiting it, you are immediately executing the next lines synchronously.

Reply:
That makes sense! Though wouldn't having `await` defeat the purpose of having an up-to-date `isLoading` value if the component is only going to be re-rendered after the promise returns?

I imagined it like this:
- update `isLoading` -> rerender (i.e. immediately executing next lines)
- when promise is resolved -> update `isLoading` -> rerender

and re: messing up the next lines: maybe something [like this](https://gist.github.com/regexyl/958aa74f8c77e6ae0176a311b407abd4#file-suggestion-ts-L255) might work?

Might be missing a bit of context and need a bit of hand-holding here